pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1047,30 +1043,14 @@ func PublishAppInfoToZedCloud(ctx *zedagentContext, uuid string,
 	}
 	statusUrl := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "info")
 
-	zedcloud.RemoveDeferred(zedcloudCtx, uuid)
 	buf := bytes.NewBuffer(data)
 	if buf == nil {
 		log.Fatal("malloc error")
 	}
 	size := int64(proto.Size(ReportInfo))
-	start := time.Now()
-	err = SendProtobuf(statusUrl, buf, size, iteration)
-	if err != nil {
-		log.Errorf("PublishAppInfoToZedCloud failed: %s", err)
-		// Try sending later
-		// The buf might have been consumed
-		buf := bytes.NewBuffer(data)
-		if buf == nil {
-			log.Fatal("malloc error")
-		}
-		zedcloud.SetDeferred(zedcloudCtx, uuid, buf, size, statusUrl,
-			true)
-	} else {
-		writeSentAppInfoProtoMessage(data)
-
-		log.Functionf("sent app info %s state %s err %t took %v",
-			uuid, state.String(), objErr, time.Since(start))
-	}
+	zedcloud.SetDeferred(zedcloudCtx, uuid, buf, size, statusUrl,
```

Comment:
It would make sense to add a comment here to the effect
 We queue the message and then get the highest priority message to send. If there are no failures and defers we'll send this message, but if there is a queue we'll retry sending the highest priority message.

That comment makes sense in all places where you have this pattern because it isn't obvious.

Reply:
Added, thank you. Sure, it is not so clear without comment.